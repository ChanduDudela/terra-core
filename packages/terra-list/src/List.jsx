import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './List.module.scss';
import Item from './ListItem';
import Section from './ListSection';
import Subsection from './ListSubsection';
import SectionHeader from './ListSectionHeader';
import SubsectionHeader from './ListSubsectionHeader';
import Utils from './ListUtils';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The children list items passed to the component.
   */
  children: PropTypes.node,
  /**
   * Whether or not the list's child items should have a border color applied.
   */
  isDivided: PropTypes.bool,
  /**
   * Function callback for the ref of the ul.
   */
  refCallback: PropTypes.func,
  /**
   * Accessibility role of the list, defaults to 'listbox'. If creating a read-only list, pass 'none', and no role will be added.
   */
  role: PropTypes.string,
};

const defaultProps = {
  children: [],
  isDivided: false,
  role: 'listbox',
};

const List = ({
  children,
  isDivided,
  refCallback,
  role,
  ...customProps
}) => {
  const listClassNames = cx([
    'list',
    customProps.className,
  ]);

  const attrSpread = {};
  if (role && role.length > 0 && role !== 'none') {
    attrSpread.role = role;
  }
  if (isDivided) {
    attrSpread['data-list-items-divided'] = isDivided;
  }

  return (
    <ul {...customProps} {...attrSpread} className={listClassNames} ref={refCallback}>
      {children}
    </ul>
  );
};

List.propTypes = propTypes;
List.defaultProps = defaultProps;

export default List;
export {
  Item,
  Section,
  Subsection,
  SectionHeader,
  SubsectionHeader,
  Utils,
};
