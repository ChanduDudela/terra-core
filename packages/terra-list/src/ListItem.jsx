import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import ChevronRight from 'terra-icon/lib/icon/IconChevronRight';
import ListUtils from './ListUtils';
import styles from './ListItem.module.scss';

const cx = classNames.bind(styles);

// Disable this lint rule to allow for the metaData prop
/* eslint-disable react/forbid-prop-types */
const propTypes = {
  /**
   * The content element to be placed inside the list item for display.
   */
  children: PropTypes.node,
  /**
   * Whether or not the list item has a disclosure indicator presented.
   */
  hasChevron: PropTypes.bool,
  /**
   * Whether or not the list item should have a border color applied.
   */
  isDivided: PropTypes.bool,
  /**
   * Whether or not the list item should have selection styles applied.
   */
  isSelected: PropTypes.bool,
  /**
   * Whether or not the list item should have styles to indicate the item is selectable.
   */
  isSelectable: PropTypes.bool,
  /**
   * The associated metaData to be provided in the onSelect callback.
   */
  metaData: PropTypes.object,
  /**
   * Function callback for when the appropriate click or key action is performed.
   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
   */
  onSelect: PropTypes.func,
  /**
   * Function callback for the ref of the li.
   */
  refCallback: PropTypes.func,
};
/* eslint-enable react/forbid-prop-types */

const defaultProps = {
  children: [],
  hasChevron: undefined,
  isDivided: false,
  isSelected: false,
  isSelectable: undefined,
};

const ListItem = ({
  children,
  hasChevron,
  isDivided,
  isSelected,
  isSelectable,
  metaData,
  onBlur,
  onClick,
  onKeyDown,
  onMouseDown,
  onSelect,
  refCallback,
  ...customProps
}) => {
  const listItemClassNames = cx([
    'list-item',
    { divided: isDivided },
    { selected: isSelected },
    { 'is-selectable': isSelectable },
    { 'item-has-chevron': hasChevron },
    customProps.className,
  ]);

  const ariaSpread = {};
  if (isSelectable) {
    ariaSpread.onClick = ListUtils.wrappedOnClickForItem(onClick, onSelect, metaData);
    ariaSpread.onKeyDown = ListUtils.wrappedOnKeyDownForItem(onKeyDown, onSelect, metaData);
    ariaSpread.tabIndex = '0';
    ariaSpread.role = 'option';
    ariaSpread['aria-selected'] = isSelected;
    ariaSpread['data-item-show-focus'] = 'true';
    ariaSpread.onBlur = ListUtils.wrappedOnBlur(onBlur, event => event.currentTarget.setAttribute('data-item-show-focus', 'true'));
    ariaSpread.onMouseDown = ListUtils.wrappedOnMouseDown(onMouseDown, event => event.currentTarget.setAttribute('data-item-show-focus', 'false'));
  }

  let childContent = children;
  if (hasChevron) {
    childContent = [
      <div className={cx('item-fill')} key="item-fill">{childContent}</div>,
      <div className={cx('item-end')} key="item-end">{<span className={cx('chevron')}><ChevronRight height="0.8em" width="0.8em" /></span>}</div>,
    ];
  }

  return (
    <li {...customProps} {...ariaSpread} className={listItemClassNames} ref={refCallback}>
      {childContent}
    </li>
  );
};

ListItem.propTypes = propTypes;
ListItem.defaultProps = defaultProps;

export default ListItem;
