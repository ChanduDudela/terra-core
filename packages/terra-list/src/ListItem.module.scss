// List Item Component
// ========================================================
:local {
  .list-item {
    background-color: var(--terra-list-v3-item-background-color);
    margin: 0;
    padding: 0;
    width: 100%;
  }

  [data-list-items-divided='true'] {
    > .list-item {
      border-top: var(--terra-list-v3-item-divider-border-top, 1px solid #dedfe0);

      &:last-of-type {
        border-bottom: var(--terra-list-v3-item-divider-border-bottom, 1px solid #dedfe0);
      }
    }

    > .selected {
      border-color: var(--terra-list-v3-item-selected-divider-border-color, #0092e0);
  
      // stylelint-disable-next-line selector-max-compound-selectors
      + .list-item {
        border-top-color: var(--terra-list-v3-item-selected-divider-border-top-color, #0092e0);
      }
    }
  }

  .chevron {
    display: flex;
    padding-left: 10px;
    padding-right: 10px;

    > svg {
      color: var(--terra-list-v3-item-chevron-color, #bcbfc0);
    }
  }

  .selected {
    background-color: var(--terra-list-v3-item-selected-background-color, darken(#ebf6fd, 2%));
  }

  .is-selectable {
    cursor: pointer;
    outline: var(--terra-list-v3-item-focus-outline, none);

    &:hover,
    &:active {
      background-color: var(--terra-list-v3-item-hover-background-color, lighten(#ebf6fd, 2%));
    }

    &[data-item-show-focus='true']:focus {
      background-color: var(--terra-list-v3-item-focus-background-color, lighten(#ebf6fd, 2%));
      box-shadow: var(--terra-list-v3-item-focus-box-shadow, inset 0 0 0 1px #26a2e5, inset 0 0 0 4px rgba(76, 178, 233, 0.2));
      z-index: 1;

      &.selected {
        background-color: var(--terra-list-v3-item-selected-focus-background-color, darken(#ebf6fd, 7%));
      }
    }

    &.selected:hover {
      background-color: var(--terra-list-v3-item-selected-hover-background-color, darken(#ebf6fd, 7%)) !important;
    }
  }

  .item-has-chevron {
    align-items: center;
    display: flex;
    justify-content: space-between;
    width: 100%;
  }

  .item-fill {
    flex: 1 1 auto;
    min-width: 0;
    width: 100%;
  }

  .item-end {
    flex: 0 0 auto;
  }
}
