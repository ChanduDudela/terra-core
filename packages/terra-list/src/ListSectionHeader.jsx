import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import ListUtils from './ListUtils';
import styles from './ListSectionHeader.module.scss';

const cx = classNames.bind(styles);

/* eslint-disable react/forbid-prop-types */
const propTypes = {
  /**
   * Whether or not the section is collapsed.
   */
  isCollapsed: PropTypes.bool,
  /**
   * Whether or not the section can be collapsed.
   */
  isCollapsible: PropTypes.bool,
  /**
   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
   */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  /**
   * The associated metaData to be provided in the onSelect callback.
   */
  metaData: PropTypes.object,
  /**
   * Function callback for when the appropriate click or key action is performed.
   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
   */
  onSelect: PropTypes.func,
  /**
   * Function callback passthrough for the ref of the section li.
   */
  refCallback: PropTypes.func,
  /**
   * Header to be placed within a section list item
   */
  title: PropTypes.string,
};
/* eslint-enable react/forbid-prop-types */

const defaultProps = {
  isCollapsed: false,
  isCollapsible: false,
  level: 2,
  title: '',
};

const SectionHeader = ({
  isCollapsed,
  isCollapsible,
  level,
  metaData,
  onBlur,
  onClick,
  onKeyDown,
  onMouseDown,
  onSelect,
  refCallback,
  title,
  ...customProps
}) => {
  const sectionHeaderClassNames = cx([
    'section-header',
    { 'is-collapsible': isCollapsible },
    customProps.className,
  ]);

  const ariaSpread = {};
  const Element = `h${level}`;
  let titleElement = <Element className={cx('title')}>{title}</Element>;
  let accordionIcon;
  if (isCollapsible) {
    accordionIcon = (
      <div className={cx('start')}>
        <span className={cx(['accordion-icon', { 'is-open': !isCollapsed }])} />
      </div>
    );
    titleElement = (
      <div className={cx('fill')}>
        {titleElement}
      </div>
    );

    ariaSpread.onClick = ListUtils.wrappedOnClickForItem(onClick, onSelect, metaData);
    ariaSpread.onKeyDown = ListUtils.wrappedOnKeyDownForItem(onKeyDown, onSelect, metaData);
    ariaSpread.tabIndex = '0';
    ariaSpread.role = 'heading';
    ariaSpread['aria-expanded'] = !isCollapsed;
    ariaSpread['aria-level'] = 1;
    ariaSpread.onBlur = ListUtils.wrappedOnBlur(onBlur, event => event.currentTarget.setAttribute('data-focusable', 'true'));
    ariaSpread.onMouseDown = ListUtils.wrappedOnMouseDown(onMouseDown, event => event.currentTarget.setAttribute('data-focusable', 'false'));
  }

  return (
    <li {...customProps} {...ariaSpread} className={sectionHeaderClassNames} ref={refCallback}>
      {accordionIcon}
      {titleElement}
    </li>
  );
};

SectionHeader.propTypes = propTypes;
SectionHeader.defaultProps = defaultProps;

export default SectionHeader;
