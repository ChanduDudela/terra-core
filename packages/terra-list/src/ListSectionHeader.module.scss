@import '~terra-mixins';

// Section Header
// ========================================================
:local {
  .section-header {
    background-color: #e8e9ea;
    border-bottom: 1px solid #c8cacb;
    padding-bottom: 0.5rem;
    padding-left: 0.7142857143rem;
    padding-top: 0.5rem;
    width: 100%;
  }

  .title {
    color: #1c1f21;
    font-size: 1rem;
    font-size: var(--terra-section-header-font-size, 1rem);
    font-weight: var(--terra-section-header-font-weight, bold);
    line-height: var(--terra-section-header-line-height, 1.142857143);
    margin: 0;
    padding: 0;
    width: 100%;
    word-break: break-all;
    word-wrap: normal;
  }

  .is-collapsible {
    align-items: center;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    padding-left: 0.5rem;
    width: 100%;

    &.section-header:focus {
      background-color: var(--terra-list-item-focus-background-color, #e8e9ea);
      box-shadow: var(--terra-list-section-header-focus-box-shadow, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35));
      outline: var(--terra-list-section-header-focus-outline, none);
      position: relative;
      z-index: 1;
    }
  
    &.section-header:hover,
    &.section-header:active {
      background-color: #d7d8d9;
    }

    &[data-focusable='false']:focus {
      box-shadow: none !important;
    }
  }

  .item-fill {
    flex: 1 1 auto;
    min-width: 0;
    width: 100%;
  }

  .item-start {
    flex: 0 0 auto;
    padding-right: 0.3571228571rem;
  }

  .accordion-icon {
    background: var(--terra-list-section-header-accordion-icon, inline-svg('<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48 48" enable-background="new 0 0 48 48" xml:space="preserve"><title>0023_caretRight_a</title><path id="caretRight" d="M12,0l24,24L12,48V0z"/></svg>'));
    background-repeat: no-repeat;
    background-size: contain;
    display: block;
    height: 0.8571428571rem;
    transform: rotate(0deg);
    width: 0.8571428571rem;

    // flips the icon about its vertical axis, presenting it in its mirrored-image form
    [dir=rtl] & {
      transform: scaleX(-1);
    }

    // tells the rtl postcss plugin to not transform this by default
    [dir] &.is-open {
      transform: rotate(90deg);
    }
  }
}
