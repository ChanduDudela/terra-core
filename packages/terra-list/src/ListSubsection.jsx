import React from 'react';
import PropTypes from 'prop-types';
import 'terra-base/lib/baseStyles';
import SubsectionHeader from './ListSubsectionHeader';

// Disable this lint rule to allow for the metaData prop
/* eslint-disable react/forbid-prop-types */
const propTypes = {
  /**
   * The children list items passed to the component.
   */
  children: PropTypes.node,
  /**
   * Whether or not the section is collapsed.
   */
  isCollapsed: PropTypes.bool,
  /**
   * Whether or not the section can be collapsed.
   */
  isCollapsible: PropTypes.bool,
  /**
   * Whether or not the header list item should have a border color applied.
   */
  isDivided: PropTypes.bool,
  /**
   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
   */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  /**
   * Function callback for the ref of the li.
   */
  metaData: PropTypes.object,
  /**
   * Function callback for the ref of the li.
   */
  onClick: PropTypes.func,
  /**
   * Function callback for the ref of the li.
   */
  onKeyDown: PropTypes.func,
  /**
   * Function callback for the ref of the li.
   */
  onSelect: PropTypes.func,
  /**
   * Function callback passthrough for the ref of the section li.
   */
  refCallback: PropTypes.func,
  /**
   * Header to be placed within a section list item
   */
  title: PropTypes.string,
};
/* eslint-enable react/forbid-prop-types */

const defaultProps = {
  children: [],
  isCollapsed: false,
  isCollapsible: false,
  isDivided: false,
  level: 2,
  title: '',
};

const ListSubsection = ({
  children,
  isCollapsed,
  isCollapsible,
  ...customProps
}) => {
  let sectionItems;
  if (!isCollapsible || !isCollapsed) {
    sectionItems = children;
  }

  return (
    <React.Fragment>
      <SubsectionHeader isCollapsible={isCollapsible} isCollapsed={isCollapsed} {...customProps} />
      {sectionItems}
    </React.Fragment>
  );
};

ListSubsection.propTypes = propTypes;
ListSubsection.defaultProps = defaultProps;

export default ListSubsection;
