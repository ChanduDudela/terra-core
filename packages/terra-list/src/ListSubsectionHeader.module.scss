@import '~terra-mixins';

// Subsection Header
// ========================================================
:local {
  .subsection-header {
    background-color: var(--terra-list-v3-subsection-header-background-color, #c8cacb);
    border-bottom: var(--terra-list-v3-subsection-header-border-bottom, 1px solid #b5b6b7);
    padding-bottom: var(--terra-list-v3-subsection-header-padding-bottom, 0.5rem);
    padding-left: var(--terra-list-v3-subsection-header-padding-left, 1.6571428571rem);
    padding-top: var(--terra-list-v3-subsection-header-padding-top, 0.5rem);
    width: 100%;
  }

  .title {
    color: var(--terra-list-v3-subsection-header-title-color, #1c1f21);
    font-size: var(--terra-list-v3-subsection-header-title-font-size, 1rem);
    font-weight: var(--terra-list-v3-subsection-header-title-font-weight, bold);
    line-height: var(--terra-list-v3-subsection-header-title-line-height, 1.142857143);
    margin: 0;
    padding: 0;
    width: 100%;
    word-break: break-all;
    word-wrap: normal;
  }

  .is-collapsible {
    align-items: center;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    padding-left: var(--terra-list-v3-subsection-header-collapsible-padding-left, 1.4471428571rem);
    width: 100%;

    &.subsection-header:focus {
      // background-color: var(--terra-list-item-focus-background-color, lighten(#ebf6fd, 2%)) !important;
      box-shadow: var(--terra-list-v3-subsection-header-focus-box-shadow, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35));
      outline: var(--terra-list-v3-subsection-header-focus-outline, none);
      position: relative;
      z-index: 1;
    }
  
    &.subsection-header:hover,
    &.subsection-header:active {
      background-color: var(--terra-list-v3-subsection-header-hover-background-color, #b5b6b7); // Needs proper color value.
    }

    &[data-focusable='false']:focus {
      box-shadow: none !important;
    }
  }

  .fill {
    flex: 1 1 auto;
    min-width: 0;
    width: 100%;
  }

  .start {
    flex: 0 0 auto;
    padding-right: var(--terra-list-v3-subsection-header-start-padding-right, 0.3571228571rem);
  }

  .accordion-icon {
    background: var(--terra-list-v3-subsection-header-accordion-icon-background, inline-svg('<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48 48" enable-background="new 0 0 48 48" xml:space="preserve"><title>0023_caretRight_a</title><path id="caretRight" d="M12,0l24,24L12,48V0z"/></svg>'));
    background-repeat: no-repeat;
    background-size: contain;
    display: block;
    height: 0.8571428571rem;
    transform: rotate(0deg);
    width: 0.8571428571rem;

    // flips the icon about its vertical axis, presenting it in its mirrored-image form
    [dir=rtl] & {
      transform: scaleX(-1);
    }

    // tells the rtl postcss plugin to not transform this by default
    [dir] &.is-open {
      transform: rotate(90deg);
    }
  }
}
