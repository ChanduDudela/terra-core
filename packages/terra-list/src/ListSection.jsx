import React from 'react';
import PropTypes from 'prop-types';
import 'terra-base/lib/baseStyles';
import SelectableUtils from './SelectableUtils';
import ListItem from './ListItem';

const propTypes = {
  /**
   * The children list items passed to the component.
   */
  children: PropTypes.node,
  /**
   * Whether or not the list item has a disclosure indicator presented.
   */
  hasChevron: PropTypes.bool,
  /**
   * Header to be placed within a section list item
   */
  header: PropTypes.element,
  /**
   * Whether or not the section is collapsed.
   */
  isCollapsed: PropTypes.bool,
  /**
   * Whether or not the list item should have selection styles applied.
   */
  isSelected: PropTypes.bool,
  /**
   * Whether or not the list item should have styles to indicate the item is selectable.
   */
  isSelectable: PropTypes.bool,
  /**
   * Function callback for the ref of the li.
   */
  listKey: PropTypes.string.isRequired,
  /**
   * Function callback for the ref of the li.
   */
  onChange: PropTypes.func,
  /**
   * Function callback for the ref of the li.
   */
  refCallback: PropTypes.func,
  /**
   * Array of keys associated to selected list items
   */
  selectedKeys: PropTypes.arrayOf(PropTypes.string),
  /**
   * Title to be applied to the section header.
   */
  title: PropTypes.string,
};

const defaultProps = {
  children: [],
  hasChevron: undefined,
  isCollapsed: false,
  isSelected: false,
  isSelectable: undefined,
  selectedKeys: [],
};

class ListSection extends React.Component {
  constructor(props) {
    super(props);
    this.aggregateRef = this.aggregateRef.bind(this);
    this.sendRefs = this.sendRefs.bind(this);
    this.aggregatedRefs = [];
  }

  componentDidMount() {
    this.sendRefs();
  }

  componentDidUpdate() {
    this.sendRefs();
  }

  sendRefs() {
    if (this.props.refCallback) {
      const tempRefs = this.aggregatedRefs;
      this.aggregatedRefs = [];
      this.props.refCallback(tempRefs);
    }
  }

  aggregateRef(node, index) {
    this.aggregatedRefs[index + 1] = node;
  }

  render() {
    const {
      children,
      disableUnselectedItems,
      hasChevron,
      header,
      isCollapsed,
      isSelected,
      isSelectable,
      listKey,
      onChange,
      refCallback,
      selectedKeys,
      ...customProps
    } = this.props;

    let sectionItems;
    if (!isCollapsed) {
      sectionItems = React.Children.map(children, (child, index) => {
        const newProps = SelectableUtils.newPropsForItem(child, onChange, hasChevron, selectedKeys, disableUnselectedItems, isSelectable);
        newProps.refCallback = node => this.aggregateRef(node, index);
        return React.cloneElement(child, newProps);
      });
    } else if (this.aggregatedRefs.length > 1) {
      this.aggregatedRefs = this.aggregatedRefs.slice(0, 1);
    }

    return (
      <React.Fragment>
        <ListItem {...customProps} refCallback={node => this.aggregateRef(node, -1)}>
          {header}
        </ListItem>
        {sectionItems}
      </React.Fragment>
    );
  }
}

ListSection.propTypes = propTypes;
ListSection.defaultProps = defaultProps;

export default ListSection;
