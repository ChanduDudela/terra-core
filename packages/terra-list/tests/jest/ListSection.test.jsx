import React from 'react';
import { Item, Section } from '../../src/index';

// Snapshot Tests
it('should render with items', () => {
  const item1 = <Item key="123" />;
  const item2 = <Item key="124" />;
  const item3 = <Item key="125" />;
  const item4 = <Item key="126" />;
  const item5 = <Item key="127" />;
  const items = [item1, item2, item3, item4, item5];
  const shallowComponent = shallow(<Section>{items}</Section>);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with no items', () => {
  const shallowComponent = shallow(<Section />);
  expect(shallowComponent).toMatchSnapshot();
});


it('should render with title', () => {
  const shallowComponent = shallow(<Section title="Test Title" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with title and level', () => {
  const shallowComponent = shallow(<Section title="Test Title" level={3} />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with isCollapsed', () => {
  const shallowComponent = shallow(<Section isCollapsed />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with isCollapsible', () => {
  const shallowComponent = shallow(<Section isCollapsible />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with callback functions', () => {
  const shallowComponent = shallow(<Section refCallback={jest.fn()} onSelect={jest.fn()} />);
  expect(shallowComponent).toMatchSnapshot();
});
