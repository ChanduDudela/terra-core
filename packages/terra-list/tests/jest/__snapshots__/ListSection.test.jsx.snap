// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with callback functions 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    onSelect={[MockFunction]}
    refCallback={[MockFunction]}
    title=""
  />
</Fragment>
`;

exports[`should render with isCollapsed 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={true}
    isCollapsible={false}
    level={2}
    title=""
  />
</Fragment>
`;

exports[`should render with isCollapsible 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={true}
    level={2}
    title=""
  />
</Fragment>
`;

exports[`should render with items 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    title=""
  />
  <ListItem
    isSelected={false}
    key="123"
  />
  <ListItem
    isSelected={false}
    key="124"
  />
  <ListItem
    isSelected={false}
    key="125"
  />
  <ListItem
    isSelected={false}
    key="126"
  />
  <ListItem
    isSelected={false}
    key="127"
  />
</Fragment>
`;

exports[`should render with no items 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    title=""
  />
</Fragment>
`;

exports[`should render with title 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    title="Test Title"
  />
</Fragment>
`;

exports[`should render with title and level 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={3}
    title="Test Title"
  />
</Fragment>
`;
